x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStepTests.java

```diff
@@ -401,4 +467,8 @@ protected <Request extends ActionRequest, Response extends ActionResponse> void
             }
         };
     }
+
+    private Integer randomTotalShardsPerNode() {
+        return randomIntBetween(-1, 100);
```

Comment:
I would suggest encapsulating the nullability (and fixing the bounds) with:
```suggestion
        return randomBoolean() ? null : randomIntBetween(1, 100);
```